build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/endpoint_parameters_class.rb

```diff
@@ -9,14 +9,20 @@ def initialize(options)
         @service = options.fetch(:service)
         if (parameters = @service.endpoint_rules&.fetch('parameters'))
           @parameters = parameters.map do |k,p|
-            EndpointParameter.new(k, p)
+            value, source = EndpointParameter.endpoint_parameter_value(@service,  k, p)
+            EndpointParameter.new(k, p, value, source)
```

Comment:
This is weird, using a class method to construct an instance of that class. Can new() just take the service, k, and p?

Reply:
Yeah, that makes sense.  Orig was trying to match old usage from both classes that used different endpointParameter classes. 

Updated.